File: Client/src/client/events/discordAuth.js
/// <reference types="@altv/types-client" />
/// <reference types="@altv/types-natives" />

import alt from 'alt-client';
import native from 'natives';

let loginHud;
let cam;
const DISCORD_APP_ID = '1102181838484668476';

alt.onServer('Client:Auth:Open', () => {
    loginHud = new alt.WebView("http://resource/client/webview/login/index.html");
    loginHud.on('AuthDiscord', getOAuthToken);
    loginHud.focus();

    alt.showCursor(true);
    alt.toggleGameControls(false);
    alt.toggleVoiceControls(false);
    native.displayRadar(false);

    cam = native.createCamWithParams('DEFAULT_SCRIPTED_CAMERA', -1373.9209, -1116.0791, 21.29309, 0, 0,  130, 90, true, 0);
    native.setCamActive(cam, true);
    native.renderScriptCams(true, false, 0, true, false, 0);
    native.freezeEntityPosition(alt.Player.local.scriptID, true);
});

alt.onServer('Client:Auth:CloseLoginHud', () => {
    alt.showCursor(false);
    alt.toggleGameControls(true);
    alt.toggleVoiceControls(true);
    alt.beginScaleformMovieMethodMinimap('SETUP_HEALTH_ARMOUR');
    native.displayRadar(true);
    native.scaleformMovieMethodAddParamInt(3);
    native.endScaleformMovieMethod();
    native.freezeEntityPosition(alt.Player.local.scriptID, false);
    native.destroyAllCams(true);
    native.renderScriptCams(false, false, 0, false, false, 0);

   if (loginHud) {
       loginHud.destroy();
   }
});

async function getOAuthToken() {
    try {
        const token = await alt.Discord.requestOAuth2Token(DISCORD_APP_ID);
        alt.emitServer('DiscordToken', token);
    } catch (e) {
        alt.logError("[Client] Es gab einen Fehler mit dem Discord Token: " + e);
    }
}

File: Los Angeles Life Client/Los Angeles Life Client/System/DiscordAuth.cs
using AltV.Net.Client;
using AltV.Net.Client.Elements.Interfaces;

namespace Los_Angeles_Life_Client.System;

public abstract class DiscordAuth : Client
{
    private const string DiscordAppId = "1102181838484668476";
    private static IWebView? _loginHud;

    public static void Load()
    {
        Alt.OnServer("Client:Auth:Open", () =>
        {
            _loginHud = Alt.CreateWebView("http://resource/net6.0/webview/login/index.html");
            _loginHud.On("AuthDiscord", GetOAuthToken);
            _loginHud.Focus();

            Alt.ShowCursor(true);
            Alt.GameControlsEnabled = false;
            Alt.VoiceControlsEnabled = false;
            Alt.Natives.DisplayRadar(false);

            var cam = Alt.Natives.CreateCamWithParams("DEFAULT_SCRIPTED_CAMERA", (float)-1373.9209, (float)-1116.0791,
                (float)21.29309, 0, 0, 130, 90, true, 0);
            Alt.Natives.SetCamActive(cam, true);
            Alt.Natives.RenderScriptCams(true, false, 0, true, false, 0);
            Alt.Natives.FreezeEntityPosition(Alt.LocalPlayer.ScriptId, true);
        });

        Alt.OnServer("Client:Auth:CloseLoginHud", () =>
        {
            Alt.ShowCursor(false);
            Alt.GameControlsEnabled = true;
            Alt.VoiceControlsEnabled = true;
            Alt.BeginScaleformMovieMethodMinimap("SETUP_HEALTH_ARMOUR");
            Alt.Natives.DisplayRadar(true);
            Alt.Natives.ScaleformMovieMethodAddParamInt(3);
            Alt.Natives.EndScaleformMovieMethod();
            Alt.Natives.FreezeEntityPosition(Alt.LocalPlayer.ScriptId, false);
            Alt.Natives.DestroyAllCams(true);
            Alt.Natives.RenderScriptCams(false, false, 0, false, false, 0);

            _loginHud?.Destroy();
        });
        
        async void GetOAuthToken()
        {
            try
            {
                var token = await Alt.Discord.RequestOAuth2Token(DiscordAppId);
                Alt.EmitServer("DiscordToken", token);
            }
            catch (Exception ex)
            {
                Alt.LogInfo("Fehler mit der Discord Auth: " + ex);
                throw;
            }
        }
    }
}